neo/Network/P2P/Payloads/BlockBase.cs

```diff
@@ -32,12 +36,23 @@ public UInt256 Hash
             }
         }
 
-        public virtual int Size => sizeof(uint) + PrevHash.Size + MerkleRoot.Size + sizeof(uint) + sizeof(uint) + NextConsensus.Size + Witness.Size;
+        public virtual int Size =>
+            sizeof(uint) +              //Version
+            PrevHash.Size +             //PrevHash
+            MerkleRoot.Size +           //MerkleRoot
+            sizeof(uint) +              //Timestamp
+            sizeof(uint) +              //Index
+            NextConsensus.Size +        //NextConsensus
+            Validators.GetVarSize() +   //Validators
+            64 * Signatures.Length;     //Witness
 
         public virtual void Deserialize(BinaryReader reader)
         {
             ((IVerifiable)this).DeserializeUnsigned(reader);
-            Witness = reader.ReadSerializable<Witness>();
+            Validators = reader.ReadSerializableArray<ECPoint>(Blockchain.MaxValidators);
+            Signatures = new byte[Validators.Length - (Validators.Length - 1) / 3][];
```

Comment:
Signatures: `2f + 1`, Validators: `3f + 1.

Reply:
Maybe we should check that you have `(3f+1)` Validators for throw a `FormatException`

Reply:
https://github.com/neo-project/neo/blob/5a658d17f247cdc01e8a6abaa68f39b380300191/neo/Network/P2P/Payloads/BlockBase.cs#L108

It is checked here.

Reply:
@erikzhang perhaps we could create a `MultiSignatures` class. So, when it's a pack of sigs like this, it works. And when it migrates to Schnorr it keeps compatibility.
It could be part of ABI `Signature`, that could be single, multi, etc.